include/net/mqtt.h

```diff
@@ -0,0 +1,629 @@
+ * @param[in] param Identifies message being acknowledged.
+ *
+ * @retval 0 or an error code indicating reason for failure.
+ *
+ * @note Default protocol revision used for connection request is 3.1.0. Please
+ *       set client.protocol_version = MQTT_VERSION_3_1_1 to use protocol 3.1.1.
+ */
+int mqtt_publish_receive(struct mqtt_client *const client,
+			 struct mqtt_pubrec_param const *const param);
+
+/**
+ * @brief API to used by publishing client to request releasing published data.
+ *        Shall be used only after @ref MQTT_EVT_PUBREC is received and is valid
+ *        only for QoS level @ref MQTT_QOS_2_EXACTLY_ONCE.
+ *
+ * @param[in] client Client instance for which the procedure is requested.
+ *                   Shall not be NULL.
+ * @param[in] param Identifies message being released.
+ *
+ * @retval 0 or an error code indicating reason for failure.
+ *
+ * @note Default protocol revision used for connection request is 3.1.0. Please
+ *       set client.protocol_version = MQTT_VERSION_3_1_1 to use protocol 3.1.1.
+ */
+int mqtt_publish_release(struct mqtt_client *const client,
+			 struct mqtt_pubrel_param const *const param);
+
+/**
+ * @brief API used by subscribing clients to acknowledge reception of a
+ *        released message. Should be used on reception @ref MQTT_EVT_PUBREL
+ *        event.
+ *
+ * @param[in] client Identifies client instance for which the procedure is
+ *                   requested. Shall not be NULL.
+ * @param[in] param Identifies message being completed.
+ *
+ * @retval 0 or an error code indicating reason for failure.
+ *
+ * @note Default protocol revision used for connection request is 3.1.0. Please
+ *       set client.protocol_version = MQTT_VERSION_3_1_1 to use protocol 3.1.1.
+ */
+int mqtt_publish_complete(struct mqtt_client *const client,
+			  struct mqtt_pubcomp_param const *const param);
+
+/**
+ * @brief API to request subscribing to a topic on the connection.
+ *
+ * @param[in] client Identifies client instance for which the procedure
+ *                   is requested. Shall not be NULL.
+ * @param[in] param Subscription parameters. Shall not be NULL.
+ *
+ * @retval 0 or an error code indicating reason for failure.
+ */
+int mqtt_subscribe(struct mqtt_client *const client,
+		   struct mqtt_subscription_list const *const param);
+
+/**
+ * @brief API to request unsubscribe from a topic on the connection.
+ *
+ * @param[in] client Identifies client instance for which the procedure is
+ *                   requested. Shall not be NULL.
+ * @param[in] param Parameters describing topics being unsubscribed from.
+ *                  Shall not be NULL.
+ *
+ * @note QoS included in topic description is unused in this API.
+ *
+ * @retval 0 or an error code indicating reason for failure.
+ */
+int mqtt_unsubscribe(struct mqtt_client *const client,
+		     struct mqtt_subscription_list const *const param);
+
+/**
+ * @brief API to send MQTT ping.
+ *
```

Comment:
Perhaps, mention that this is optional, and lib will send it itself based on interval specified in ...

Reply:
(xref to mqtt_live()).